packages/editor/src/store/actions.js

```diff
@@ -725,6 +727,99 @@ export function setIsListViewOpened( isOpen ) {
 	};
 }
 
+/**
+ * Action that toggles Distraction free mode.
+ * Distraction free mode expects there are no sidebars, as due to the
+ * z-index values set, you can't close sidebars.
+ */
+export const toggleDistractionFree =
+	() =>
+	( { dispatch, registry } ) => {
+		const isDistractionFree = registry
+			.select( preferencesStore )
+			.get( 'core', 'distractionFree' );
+		if ( isDistractionFree ) {
+			registry
+				.dispatch( preferencesStore )
+				.set( 'core', 'fixedToolbar', false );
```

Comment:
This should be changed in line with the rest of the PR and check isDistrationFree wherever the toolbars are rendered instead of setting the prefference.

Also if we do that the top toolbar option in the mode menu has to be disabled in DFM.

Reply:
I left this separate, it's not the only thing that should be done this way.